equinox/nn/_embedding.py

```diff
@@ -89,3 +96,313 @@ def __call__(
+
+        x_rope = (x * freqs_real) + (neg_half_x * freqs_imag)
+        return jax.lax.stop_gradient(x_rope)
+
+
+class SinusoidalPositionalEmbedding(Module):
+    r"""
+    A sinusoidal positional encoding module, as described in the paper
+    "Attention is All You Need". While this module can be used in any context, it is
+    particularly useful for providing positional information to transformer models.
+
+    !!! example
+        The following example demonstrates how to use `SinusoidalPositionalEmbedding` in
+        a simple transformer model.
+        ```python
+
+        class TransformerBlock(eqx.nn.StatefulLayer):
+            ...
+            key_embeddings: SinusoidalPositionalEmbedding
+            query_embeddings: SinusoidalPositionalEmbedding
+
+            def __init__(...):
+                ...
+                self.query_embeddings = SinusoidalPositionalEmbedding(
+                    embedding_size=n_embd
+                )
+                self.key_embeddings = SinusoidalPositionalEmbedding(
+                    embedding_size=n_embd
+                )
+                ...
+
+            def __call__(...):
+                def process_heads(query_heads, key_heads, value_heads):
+                    query_heads = jax.vmap(self.query_embeddings,
+                                           in_axes=1,
+                                           out_axes=1)(query_heads)
+                    key_heads = jax.vmap(self.key_embeddings,
+                                         in_axes=1,
+                                         out_axes=1)(key_heads)
+
+                    return query_heads, key_heads, value_heads
+
+                mha_output = self.mha_attention(
+                    process_heads=process_heads,
+                    query=jax.vmap(self.rms_norm)(x),
+                    key_=jax.vmap(self.rms_norm)(x),
+                    value=jax.vmap(self.rms_norm)(x),
+                    mask=mask,
+                )
+        ```
+
+    ??? cite
+
+        [Attention is All You Need](https://arxiv.org/abs/1706.03762)
+
+        ```bibtex
+        @inproceedings{vaswani2017attention,
+            author={Vaswani, Ashish and Shazeer, Noam and Parmar, Niki and
+                    Uszkoreit, Jakob and Jones, Llion and Gomez, Aidan N and
+                    Kaiser, {\L}ukasz and Polosukhin, Illia},
+            booktitle={Advances in Neural Information Processing Systems},
+            publisher={Curran Associates, Inc.},
+            title={Attention is All You Need},
+            volume={30},
+            year={2017}
+        }
+        ```
+    """
+
+    embedding_size: int = field(static=True)
+    theta: float = field(static=True)
+
+    def __init__(
+        self,
+        embedding_size: int,
+        *,
+        theta: float = 10000.0,
+        key: Optional[PRNGKeyArray] = None,
+        **kwargs,
+    ):
+        """**Arguments:**
+        `SinusoidalPositionalEmbedding` requires:
```

Comment:
Nit: can we remove the `` `SinusoidalPositionalEmbedding` requires:`` line from the docstring?